### PR TITLE
Bids 3162/summary chart v2 5

### PR DESCRIPTION
**Pull request by MauserBitfly**
This PR:
Implements the Summary 2.5 chart changes
- New Filter options
  - aggregation (some options only available for permium users charts *)
  - efficiency
  - groups: Multi select + 2 special options (network average and total)
 - data points now based on timestamp an no longer on epochs

This PR did not
- Test the chart with new data (the backend does not throw an error with the new options, but still returns the old data) -> currently the dates / epochs in the chart are wrong displayed as they are based on timestamps but get epochs from the backend. 
- Remove the `showInDevelopment` flag as the Chart should only be made public after tests against the new backend. 
- Implement loading of partial time frame loading. We will first see how the chart performs when loading all the data and only when needed implement partial loading. As this would decrease usability and should only be done when we get into performance issues. 
  
  
*) add. info: the permissions for the chart depend on the user how created the dashboard and not on the logged-in user - important for shared dashboards.

### CODE REVIEW

File: frontend/components/dashboard/chart/SummaryChart.vue
Comment (D13ce) [inline frontend/components/dashboard/chart/SummaryChart.vue:1]: **Suggestion:** This is not related to this PR, but the tooltip should probably only show two decimal points, right?
image

File: frontend/components/dashboard/chart/DashboardChartSummaryChartFilter.vue
Comment (D13ce) [inline frontend/components/dashboard/chart/DashboardChartSummaryChartFilter.vue:1]: **Suggestion:** Should we maybe debounce this? So if a user changes the group settings quickly, there are not that many requests.

**Review by D13ce — CHANGES_REQUESTED** [review]
Well done. Only found minor issues, PTAL below :point_down:

File: frontend/components/dashboard/chart/SummaryChart.vue [hunk 1]
```diff
@@ -109,17 +129,16 @@ const option = computed(() => {
         fontSize: textSize,
         lineHeight: 20,
         formatter: (value: number) => {
-          const date = formatEpochToDate(value, $t('locales.date'))
-          if (date === undefined) {
-            return ''
+          const date = formatGoTimestamp(value, undefined, 'absolute', 'narrow', $t('locales.date'), false)
+          if (aggregation.value === 'epoch') {
+            return `${date}\n${$t('common.epoch')} ${tsToEpoch(value)}`
           }
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply frontend/components/dashboard/chart/SummaryChart.vue:135]: It is a Minus sign due to the data beeing epoch and not a timestamp

[the remark this answers is not in the document]
Comment (D13ce) [reply frontend/components/dashboard/chart/SummaryChart.vue:135]: Ah, of course. Thanks for the clarification, resolved.

File: frontend/components/dashboard/chart/SummaryChart.vue
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply frontend/components/dashboard/chart/SummaryChart.vue:1]: should be fixed now

Comment (MauserBitfly) [reply frontend/components/dashboard/chart/SummaryChart.vue:1]: Had the same thought, but then got distracted in the office and forgot about it ... changed ;-)

File: frontend/components/dashboard/chart/DashboardChartSummaryChartFilter.vue
Comment (MauserBitfly) [reply frontend/components/dashboard/chart/DashboardChartSummaryChartFilter.vue:1]: good point ... sending now a request on the first change (so if he only chances one item it is fast) and then debounce if he makes multiply changes

**Review by D13ce — APPROVED** [review]
LGTM

**Pull request merged**